sdk/log/logger.go

```diff
@@ -68,6 +71,12 @@ func (l *logger) newRecord(ctx context.Context, r log.Record) Record {
 		attributeValueLengthLimit: l.provider.attributeValueLengthLimit,
 		attributeCountLimit:       l.provider.attributeCountLimit,
 	}
+
+	// This field SHOULD be set once the event is observed by OpenTelemetry.
+	if newRecord.observedTimestamp.Equal(time.Time{}) {
```

Comment:
```suggestion
	if newRecord.observedTimestamp.IsZero() {
```

https://pkg.go.dev/time#Time.IsZero